pkg/codegen/schema/schema.go

```diff
@@ -1151,16 +1168,35 @@ func (pkg *Package) marshalFunction(f *Function) (FunctionSpec, error) {
 		outputs = &outs.ObjectTypeSpec
 	}
 
+	var returnType *ReturnTypeSpec
+	if f.ReturnType != nil {
+		returnType = &ReturnTypeSpec{}
+		if objectType, ok := f.ReturnType.(*ObjectType); ok {
```

Comment:
What about `"$ref"` based objects?

Reply:
I _assumed_ that `$ref`-ed objects would be generated when they are part of the schema `types` but I will add some tests to make sure

Reply:
My concern here was that they will be deserialized both as part of the return type and in the tyes section. I doesn't look like `"$ref"` objects will be left as references on the way out.

Reply:
There should definitely be a distinction: `$ref` types should be generated as they are encountered. 

My tests are showing at least the dotnet SDK-gen that:
  - Object type definitions in `types` are _not_ generated separately
  - Referenced object type definitions from the ReturnType are generated as they are encountered (should not be the case)

Reply:
Fixed this issue BTW ✅